src/Widgets/Window.vala

```diff
@@ -179,23 +197,34 @@ public class Window : Gtk.Window {
         load_window_state ();
         window_position = Gtk.WindowPosition.CENTER;
         set_hide_titlebar_when_maximized (false);
-        icon_name = MarkMyWords.ICON_NAME;
 
-        layout = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
-        layout.homogeneous = false;
+        var box = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
+        box.expand = true;
 
+        Gtk.Builder builder;
         if (prefs.use_headerbar) {
-            toolbar = new Toolbar ();
-            toolbar.set_title (MarkMyWords.APP_NAME);
-            set_titlebar (toolbar as Gtk.Widget);
+            builder = new Gtk.Builder.from_resource ("/com/voldyman/markmywords/ui/headerbar.ui");
+            headerbar = (Gtk.HeaderBar) builder.get_object ("headerbar");
+            headerbar.set_title (MarkMyWords.APP_NAME);
+
+            set_titlebar ((Gtk.Widget) headerbar);
+            add (box);
         } else {
-            var menubar = new Menubar (this);
-            toolbar = menubar;
-            layout.pack_start (menubar, false);
+            builder = new Gtk.Builder.from_resource ("/com/voldyman/markmywords/ui/toolbar.ui");
+            Gtk.Box layout = (Gtk.Box) builder.get_object ("layout");
+
+            layout.pack_start (box, false);
+            add (layout);
         }
 
-        var box = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
-        box.expand = true;
+        Gtk.IconTheme icon_theme = Gtk.IconTheme.get_default ();
+        Gtk.Image fallback1 = (Gtk.Image) builder.get_object ("fallback1");
```

Comment:
Maybe rename this to `menu-icon` or something like this? (And the next one too: `export-icon`)